File: 611-valid-triangle-number/valid-triangle-number.go
func triangleNumber(nums []int) int {
	sort.Ints(nums)
	start := 0

	for i, v := range nums {
		if v > 0 {
			start = i
			break
		}
	}

	total := 0
    
	for mid := start + 1; mid < len(nums)-1; mid++ {
		for short, long := start, mid+1; long < len(nums) && short < mid; {
			if nums[short]+nums[mid] > nums[long] {
				total += mid - short
				long++
			} else {
				short++
			}
		}
	}

	return total
}

File: 611-valid-triangle-number/valid-triangle-number.py
class Solution:
    def triangleNumber(self, nums: List[int]) -> int:
        nums.sort()
        count = 0
        n = len(nums)

        for c in range(n - 1, 1, -1):
            a, b = 0, c - 1
            while a < b:
                if nums[a] + nums[b] > nums[c]:
                    count += b - a
                    b -= 1  # Move the upper pointer down
                else:
                    a += 1  # Move the lower pointer up

        return count
